catalog/db_branch_scanner.go

```diff
@@ -35,9 +42,36 @@ func NewDBBranchScanner(tx db.Tx, branchID int64, commitID CommitID, opts *DBSca
 		s.opts.BufferSize = DBScannerDefaultBufferSize
 	}
 	s.buf = make([]*DBScannerEntry, 0, s.opts.BufferSize)
+	commitsWhere, _ := getRelevantCommitsCondition(tx, branchID, commitID)
+	s.commitsWhere = commitsWhere
 	return s
 }
 
+func getRelevantCommitsCondition(tx db.Tx, branchID int64, commitID CommitID) (string, error) {
+	var maxCommitId CommitID
+	var commits []string
+	var commitsWhere string
+	if commitID == UncommittedID || commitID == CommittedID {
+		maxCommitId = MaxCommitID
+	} else {
+		maxCommitId = commitID
+	}
+	sql := "SELECT commit_id::text FROM catalog_commits WHERE branch_id = $1 AND commit_id <= $2 ORDER BY commit_id"
```

Comment:
This changes the SQL when `commitID == CommittedID`, from checking `commit_id < MaxCommitId` to checking `commit_id <= MaxCommitId`.

Reply:
This is a query on catalog_commits. where MaxCommitId will never appear. If looking for uncomitted - then that number ia appended